src/Badges.t.sol

```diff
@@ -921,4 +924,166 @@ contract BadgesTest is Test {
     (uint8 v, bytes32 r, bytes32 s) = vm.sign(raftHolderPrivateKey, hash);
     signature = abi.encodePacked(r, s, v);
   }
+
+  function testGiveToManyHappyPath() public {
+    address active = raftHolderAddress;
+    address recipient1 = claimantAddress;
+    uint256 recipient1PrivateKey = claimantPrivateKey;
+    address recipient2 = passiveAddress;
+    uint256 recipient2PrivateKey = passivePrivateKey;
+    address recipient3 = vm.addr(randomPrivateKey);
+    uint256 recipient3PrivateKey = randomPrivateKey;
+
+    createRaftAndRegisterSpec();
+
+    bytes memory recipient1Signature = getSignature(
+      active,
+      recipient1PrivateKey
+    );
+
+    bytes memory recipient2Signature = getSignature(
+      active,
+      recipient2PrivateKey
+    );
+
+    bytes memory recipient3Signature = getSignature(
+      active,
+      recipient3PrivateKey
+    );
+
+    address[] memory recipientsAddresses = new address[](3);
+    recipientsAddresses[0] = recipient1;
+    recipientsAddresses[1] = recipient2;
+    recipientsAddresses[2] = recipient3;
+
+    bytes[] memory recipientsSignatures = new bytes[](3);
+    recipientsSignatures[0] = recipient1Signature;
+    recipientsSignatures[1] = recipient2Signature;
+    recipientsSignatures[2] = recipient3Signature;
+
+    vm.prank(active);
+    badgesWrappedProxyV1.giveToMany(
+      recipientsAddresses,
+      specUri,
+      recipientsSignatures
+    );
+
+    assertEq(badgesWrappedProxyV1.balanceOf(recipient1), 1);
+    assertEq(badgesWrappedProxyV1.balanceOf(recipient2), 1);
+    assertEq(badgesWrappedProxyV1.balanceOf(recipient3), 1);
+  }
+
+  function testGiveToManyWithOneBadSignature() public {
+    address active = raftHolderAddress;
+    address recipient1 = claimantAddress;
+    uint256 recipient1PrivateKey = claimantPrivateKey;
+    address recipient2 = passiveAddress;
+    uint256 recipient2PrivateKey = passivePrivateKey;
+    address recipient3 = vm.addr(randomPrivateKey);
+
+    createRaftAndRegisterSpec();
+
+    bytes memory recipient1Signature = getSignature(
+      active,
+      recipient1PrivateKey
+    );
+
+    bytes memory recipient2Signature = getSignature(
+      active,
+      recipient2PrivateKey
+    );
+
+    bytes memory recipient3Signature = bytes("bad signature");
+
+    address[] memory recipientsAddresses = new address[](3);
+    recipientsAddresses[0] = recipient1;
+    recipientsAddresses[1] = recipient2;
+    recipientsAddresses[2] = recipient3;
+
+    bytes[] memory recipientsSignatures = new bytes[](3);
+    recipientsSignatures[0] = recipient1Signature;
+    recipientsSignatures[1] = recipient2Signature;
+    recipientsSignatures[2] = recipient3Signature;
+
+    vm.prank(active);
+    // expect it to revert since we passed in a bad signature
+    vm.expectRevert(bytes(errInvalidSig));
```

Comment:
Here one issue is that when 1 of 100 entries fail, how do we let the caller know it was item "65" that actually failed?